projects/unrar/unrar_fuzzer.cc

```diff
@@ -0,0 +1,30 @@
+#include <memory>
+#include <stddef.h>
+#include <string>
+#include <unistd.h>
+
+#include "rar.hpp"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  char filename[] = "mytemp.XXXXXX";
+  int fd = mkstemp(filename);
+  write(fd, data, size);
```

Comment:
Is there anyway to avoid writing data into file and reading its content back? Otherwise fuzzing would be much slower than it could be if we did everything in memory.

Reply:
Unfortunately, at the moment the unrar SDK does not provide an API for providing the contents of the file as an input. I can check with the maintainer if they'd be willing to support that in the future.